src/main.rs

```diff
@@ -17,7 +18,31 @@ use tokio::sync::Mutex;
 const REGISTRY_URL:&str = "registry.opensuse.org/home/jcronenberg/migrate-wicked/containers/opensuse/migrate-wicked-git:latest";
 const TABLE_NAME: &str = "entries";
 const DEFAULT_DB_PATH: &str = "/var/lib/wicked_migration_server/db.db3";
-const FILE_EXPIRATION_IN_SEC: u64 = 5; //* 60;
+const FILE_EXPIRATION_IN_SEC: u64 = 5 * 60;
+
+enum FileType{
+    Xml,
+    Ifcfg,
+    Netconfig,
+}
+
+impl FromStr for FileType  {
+    type Err = &'static str;
+    fn from_str(file_type: &str) -> Result<Self, Self::Err>{
+        match file_type{
+            "text/xml" => Ok(FileType::Xml),
+            "text/plain" => Ok(FileType::Ifcfg),
+            "application/octet-stream" => Ok(FileType::Ifcfg),
+            _ => Err("try_from() failed")
```

Comment:
Could be a lot more helpful, e.g. something like `"Unsupported file type: {}", file_type`